etcdserver/server.go

```diff
@@ -222,6 +222,16 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		return nil, err
 	}
 
+	isSuccess := false
+	defer func() {
+		if !isSuccess {
```

Comment:
probably we can just check if error is nil. Like

https://play.golang.org/p/RKO15iSwiV